Source/IGListAdapterDataSource.h

```diff
@@ -60,7 +60,27 @@ NS_SWIFT_NAME(ListAdapterDataSource)
  adding the background view and maintaining its visibility.
  */
 - (nullable UIView *)emptyViewForListAdapter:(IGListAdapter *)listAdapter;
-
+    
+@optional
+    
+/**
+ Asks the datasource whether a list object should be movable as the result of interactive reordering.
+ 
+ @param listAdapter The list adapter sending this information.
+ @param sectionIndex The index of the section proposed to be moved.
+ @param index The index of the object proposed to be moved.
+ */
+- (BOOL)listAdapter:(IGListAdapter *)listAdapter canMoveObjectInSection:(NSInteger)sectionIndex atIndex:(NSInteger)index;
+    
+/**
+ Notifies the datasource that a list object should move as the result of interactive reordering.
+ 
+ @param listAdapter The list adapter sending this information.
+ @param sourceIndex The starting index of the object.
+ @param destinationIndex The ending index of the object.
+ */
+- (void)listAdapter:(IGListAdapter *)listAdapter moveSectionAtIndex:(NSInteger)sourceIndex toIndex:(NSInteger)destinationIndex;
```

Comment:
B/c data sources could be out of sync w/ what is being displayed (ultimately held by `IGListAdapter` internally), I think we should do one of the following:

- Add params for to/from objects
- Add to/from object _arrays_